test/inductor/test_triton_kernels.py

```diff
@@ -491,6 +492,36 @@ def call_triton(
         compiled_result = torch.compile(call_triton)(t1, t2)
         self.assertEqual(torch_result, compiled_result)
 
+    @requires_gpu
+    @common_utils.parametrize("intermediate", [False, True])
+    def test_reinplace(self, intermediate):
+        with torch.library._scoped_library("mylib", "FRAGMENT") as m:
+            m.define("foo(Tensor x, Tensor(a!) out) -> ()")
+
+            def foo(x: torch.Tensor, out: torch.Tensor) -> None:
+                out.copy_(x.sin())
+
+            m.impl("foo", foo, "CompositeExplicitAutograd")
+
+            def f(x, out):
+                if intermediate:
+                    out = torch.empty_like(x)
+                torch.ops.mylib.foo(x, out)
+                torch.ops.mylib.foo(out, out)
+                torch.ops.mylib.foo(out, out)
+                return out
```

Comment:
If we find it interesting, this is the equivalent triton kernel case: https://gist.github.com/zou3519/303b9f0600cb31d7bbb01c2a6c39c66f

Note the additional buffer (buf0).